Mac/Components/Queue/QueueViewController.swift

```diff
@@ -42,9 +42,10 @@ class QueueViewController: NSViewController {
   override func keyDown(with event: NSEvent) {
     switch event.keyCode {
     case NSEvent.keyCodeBS:
-      let queuePos = queueView.selectedRow
-
-      App.mpdClient.removeSong(at: queuePos)
+        if queueView.numberOfRows > 0 {
+            let queuePos = queueView.selectedRow
+            App.mpdClient.removeSong(at: queuePos)
+        }
```

Comment:
I'm not 100% sure that this will work as you expect. If there are any songs in the queue it will try and remove the song anyway.

You probably want to do something more like this:

```swift
guard let queuePos = queueView.selectedRow
  else { return }

App.mpdClient.removeSong(at: queuePos)
```

This will prevent the `removeSong` call from being made if there is no selected song.